app/allocations/middleware/set-results.allocations.js

```diff
@@ -7,12 +7,20 @@ const allocationService = require('../../../common/services/allocation')
 async function setResultsAllocations(req, res, next) {
   const currentLocationId = get(req.session, 'currentLocation.id')
   const userPermissions = get(req.session, 'user.permissions')
+  const { sortBy, sortDirection } = req.body.allocations
+  const currentUrl = {
+    pathname: req._parsedOriginalUrl.pathname,
+    query: req.query,
+  }
   const displayConfig = {
     showRemaining: permissions.check(
       'allocation:person:assign',
       userPermissions
     ),
     showFromLocation: !currentLocationId,
+    sortDirection,
+    sortBy,
+    currentUrl,
```

Comment:
As mentioned, this could just take the current querystring instead of the whole URL, and replace just the sort portion of the querstring